### ISSUE TITLE
Review draft quiz statement templates added to the feature-quiz branch

### ISSUE THREAD

**Issue by ds10** [jiscdev/xapi#188]
The quiz statements in feature-quiz branch need Jisc review before deciding if these are to be pulled into 1.0.

See ticket #144

**Comment by michaelwebbjisc** [comment]
I've gone through these in detail now, and looked at how they would be implemented in Moodle.

The flow doesn't quite work as it stands, and I think this is similar for Blackboard.

Viewing the quiz works as is.

Viewing the question doesn't - in Moodle the page may should one or more questions any any time, and moodle doesn't record what is actually being shown, so all we can sent is the at the quiz has been viewed.

The user then answers questions, and then submits/completes.

So I think the 'answer questions' should use:

"http://adlnet.gov/expapi/verbs/answered"

And when they complete the quiz:

http://adlnet.gov/expapi/verbs/completed

This is the sequence the current moodle plugin uses.

The other think I noted was that we might want to use:

https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#interactionacts

instead of:

' "extensions": {
      "http://xapi.jisc.ac.uk/subType": "http://moodle.xapi.jisc.ac.uk/multiple-choice"
    }'

This would be in the question answered recipe, not viewed.  I'll add in some examples shortly.

**Comment by michaelwebbjisc** [comment]
OK, here's example 1.

So in moodle, event's aren't recorded until you submit your answers.  At this point it goes in as 'questions answered'  

Here the questions is 'Greener is a color' and I've incorrectly answered 'True'.  Module gives a result so this can be completed.

```
{

  "statement": {

    "stored": "2017-08-10T14:09:02.544000+00:00",
    "context": {
      "extensions": {
        "http://xapi&46;jisc&46;ac&46;uk/courseArea": {
          "http://xapi&46;jisc&46;ac&46;uk/vle_mod_id": "Test1"
        },
        "http://id&46;tincanapi&46;com/extension/ip-address": "0:0:0:0:0:0:0:1",
        "http://xapi&46;jisc&46;ac&46;uk/version": "x-2017-05-16",
        "http://xapi&46;jisc&46;ac&46;uk/sessionId": "Iye9OqwM9O"
      },
      "platform": "Moodle",
      "contextActivities": {
        "parent": [
          {
            "definition": {
              "extensions": {
                "http://xapi&46;jisc&46;ac&46;uk/subType": "http://xapi.jisc.ac.uk/vle/quiz"
              },
              "description": {
                "en": "A module"
              },
              "name": {
                "en": "mquiz"
              },
              "type": "http://xapi.jisc.ac.uk/vle/quiz"
            },
            "id": "http://localhost/moodle/mod/quiz/view.php?id=10",
            "objectType": "Activity"
          }
        ]
      }
    },
    "actor": {
      "account": {
        "homePage": "http://localhost/moodle",
        "name": "stu1"
      },
      "objectType": "Agent"
    },
    "timestamp": "2017-08-10T16:09:01+02:00",
    "version": "1.0.0",
    "id": "7e6452e8-070b-41c1-8aa3-c1da14d28d80",
    "result": {
      "response": "true",
      "completion": true,
      "success": false,
      "score": {
        "max": 1
      }
    },
    "verb": {
      "display": {
        "en": "answered"
      },
      "id": "http://adlnet.gov/expapi/verbs/answered"
    },
    "object": {
      "definition": {
        "correctResponsesPattern": [
          "false"
        ],
        "interactionType": "true-false",
        "description": {
          "en": "Greener is a color"
        },
        "name": {
          "en": "Greener is a color"
        },
        "type": "http://adlnet.gov/expapi/activities/module"
      },
      "id": "http://localhost/moodle/mod/quiz/view.php?id=10",
      "objectType": "Activity"
    }
  },
}```

**Comment by ds10** [comment]
I'm putting these in the templates and looking through them, so I'm adding two statement templates to the branch.

The first one is your questions completed, there it mostly makes sense to me apart from the project.

The object.activity.type is module, is this correct? The object.id is http://localhost/moodle/mod/quiz/view.php?id=10 , this is the same id as the quiz in the contextactivities parent, which has type and subtype http://xapi.jisc.ac.uk/vle/quiz".

I'm guessing the module type is incorrect?  What should the type be?

Should the object and its contentActivities parent have the same id?

**Comment by arc12** [comment]
module is definitely wrong. that got in somewhere else as a bug too, was implemented by Bb (where the dev really should have spotted something was wrong), and was fixed elsewhere by Wilbert. Top of my head, it should be http://xapi.jisc.ac.uk/vle/quiz for a quiz and something else for when the statement refers to a question. (last time I checked we didn't have a URI for that type).

For a question, I would expect the parent to be the quiz id. For a quiz, parent probably the overal module/course-area. I can;t see how the same id makes sense...
